src/preamble.js

```diff
@@ -1093,26 +1095,35 @@ function createWasm() {
 
 #if expectToReceiveOnModule('instantiateWasm')
   // User shell pages can write their own Module.instantiateWasm = function(imports, successCallback) callback
-  // to manually instantiate the Wasm module themselves. This allows pages to run the instantiation parallel
-  // to any other async startup actions they are performing.
-  // Also pthreads and wasm workers initialize the wasm instance through this path.
+  // to manually instantiate the Wasm module themselves. This allows pages to
+  // run the instantiation parallel to any other async startup actions they are
+  // performing.
+  // Also pthreads and wasm workers initialize the wasm instance through this
+  // path.
   if (Module['instantiateWasm']) {
+
 #if USE_OFFSET_CONVERTER
 #if ASSERTIONS && PTHREADS
     if (ENVIRONMENT_IS_PTHREAD) {
       assert(Module['wasmOffsetData'], 'wasmOffsetData not found on Module object');
     }
 #endif
-    wasmOffsetConverter = resetPrototype(WasmOffsetConverter, Module['wasmOffsetData']);
+    if (Module['wasmOffsetData']) {
```

Comment:
`instantiateWasm` above on line 1103 is a user hook, but this is an internal thing, isn't it? If so then I think a comment would be good (and below) to explain why it could not exist. Reading the code, it looks like `worker.js` always sets it, so it should always exist?